mindmeld/core.py

```diff
@@ -792,6 +793,18 @@ def __init__(
         if self.custom_eval and not callable(custom_eval):
             raise TypeError("Invalid custom validation function type.")
 
+    def to_dict(self):
+        """Converts the entity into a dictionary"""
+        base = {"entity": self.entity}
+        for field in ["role", "responses", "retry_response", "value", "default_eval", "hints", "custom_eval"]:
```

Comment:
You could use `self.__dict__` to iterate through the writable attributes

Reply:
Thanks!